kurtosis-devnet/tests/eigenda/failover_test.go

```diff
@@ -0,0 +1,409 @@
+
+}
+
+// Test Harness, which contains all the state needed to run the tests.
+// harness also defines some higher-level "require" methods that are used in the tests.
+type harness struct {
+	enclaveCtx          *enclaves.EnclaveContext
+	endpoints           *EnclaveServiceEndpoints
+	clients             *EnclaveServiceClients
+	batchInboxAddr      common.Address
+	testStartL1BlockNum uint64
+}
+
+func newHarness(t *testing.T) *harness {
+	// We leave 20 seconds to build the entire testHarness.
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	// Create a Kurtosis context
+	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+	require.NoError(t, err)
+
+	// Get the eigenda-memstore-devnet enclave (assuming it's already running)
+	enclaveCtx, err := kurtosisCtx.GetEnclaveContext(ctxWithTimeout, enclaveName)
+	require.NoError(t, err, "Error getting enclave context: is enclave %v running?", enclaveName)
+
+	endpoints, err := getEndpointsFromKurtosis(enclaveCtx)
+	require.NoError(t, err)
+	t.Logf("Endpoints: %+v", endpoints)
+
+	clients, err := getClientsFromEndpoints(endpoints)
+	require.NoError(t, err)
+
+	// Get the batch inbox address
+	var rollupConfigMap struct {
+		BatchInboxAddress string `json:"batch_inbox_address"`
+	}
+	err = clients.opNodeClient.CallContext(ctxWithTimeout, &rollupConfigMap, "optimism_rollupConfig")
+	require.NoError(t, err)
+
+	// Get the current L1 block number
+	testStartL1BlockNum, err := clients.gethL1Client.BlockNumber(ctxWithTimeout)
+	require.NoError(t, err)
+
+	return &harness{
+		enclaveCtx:          enclaveCtx,
+		endpoints:           endpoints,
+		clients:             clients,
+		batchInboxAddr:      common.HexToAddress(rollupConfigMap.BatchInboxAddress),
+		testStartL1BlockNum: testStartL1BlockNum,
+	}
+}
+
+// requireBatcherTxsToBeFromLayer checks that the batcher transactions since startingFromBlockNum are all from the expectedLayer.
+// It allows for up to 3 initial commitments to be of the wrong type, as the failover/failback might not have taken effect yet.
+// It requires that at least 2 commitments of the expected type are present after the failover/failback.
+func (h *harness) requireBatcherTxsToBeFromLayer(t *testing.T, startingFromBlockNum uint64, expectedLayer DALayer) {
+	batcherTxs, err := fetchBatcherTxsSinceBlock(h.endpoints.GethL1Endpoint, h.batchInboxAddr.String(), startingFromBlockNum)
+	require.NoError(t, err)
+	t.Logf("Fetched %d batcher transactions since block %d", len(batcherTxs), startingFromBlockNum)
+
+	// We allow first 3 commitments to be of the wrong DA layer, as the failover/failback might not have taken effect yet.
+	wrongCommitmentsToDiscard := 0
+	for _, batcherTx := range batcherTxs {
+		if batcherTx.daLayer != expectedLayer {
+			wrongCommitmentsToDiscard++
+		}
+		// as soon as we see 3 ethDA commitments, or an EigenDA commitment, we stop
```

Comment:
doc seems to describe a previous version of the code which wasn't generalized to `expectedLayer`

Reply:
[c63c4d0](https://github.com/Layr-Labs/optimism/pull/33/commits/c63c4d0075e303c4366026a9369054bcce283249)